examples/year_playlist.py

```diff
@@ -0,0 +1,145 @@
+#!/usr/bin/env python3
+
+"""Script for retrieving songs from a user's 'Your Top Songs 2017'
+playlist.
+
+Note: Requests to Spotify API do not currently include relevant fields arg,
+which should eventually be added.
+"""
+
+from collections import namedtuple
+import sys
+import argparse
+import spotipy
+from spotipy import oauth2
+
+# Client ID and Client Secret for zachthehammer's spotipy-test API key
+_CLIENT_ID     = '5c243ed8edc9475a90bda2cc4a9a828c'
+_CLIENT_SECRET = 'b414caddc10143eea70db6edc1f4fb47'
+
+# Your Top Songs 2017 is actually owned by spotify user.
+_SPOTIFY_USER_ID    = 'spotify'
+_YEAR_PLAYLIST_NAME = 'Your Top Songs 2017'
+
+# Format strings
+_SCRIPT_DESCRIPTION_FMT = 'Get the list of tracks from a user\'s "{}" playlist.'
```

Comment:
what's with all the leading underscores? I think it python it's usually just CONSTANT_CASE

Reply:
So I only started doing it because leading underscore hides variables & functions from help. I do agree it looks a little janky though in a library I definitely would add the leading underscores to make help message hide implementation details.
```py
[_]MY_STRING = 'This is my string.'
[_]MY_NUMBER = 6

if __name__ == '__main__':
    print('My string is: "{}". My number is: {}'.format(MY_STRING, MY_NUMBER))
```
```
Help on module without_underscore:

NAME
    without_underscore

DATA
    MY_NUMBER = 6
    MY_STRING = 'This is my string.'

FILE
    /Users/zhammer/Dropbox/spotipy-test/examples/without_underscore.py
```
```
Help on module with_underscore:

NAME
    with_underscore

FILE
    /Users/zhammer/Dropbox/spotipy-test/examples/with_underscore.py

```

Reply:
what do yo mean "hides variables & functions from help"?

Reply:
what's "help"?

Reply:
From the interactive pyshell.
```
>>> import json
>>> help(json)
Help on package json:

NAME
    json

MODULE REFERENCE
    https://docs.python.org/3.6/library/json
...
```